python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,559 @@
+                    owner_name = self.io_owner.name
+
+                return owner_name, self.name
+
+            def get_owner_indx(self):
+                """return index if owner is ModuleWrapper, if not return 0"""
+                if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                    return self.io_owner.index
+
+                # if not ModuleWrapper then owner is PipelineConfig, return 0
+                # to identify this is global interface
+                return 0
+
+            def __repr__(self):
+                """Get all binding(input data), exepect like |data_0: mod1:data_0"""
+                ret = "  |{}: ".format(self.name)
+                for binding in self.bindings:
+                    mname, dname = binding.get_name()
+                    ret += "{0}:{1} ".format(mname, dname)
+                return ret
+
+            def dag_acircle_check(self, start, inputs):
+                """check if the DAG that current binding stay is acircle"""
+                for _, binding in inputs.items():
+                    if start == binding.io_owner:
+                        return False
+                    for p in binding.parents:
+                        if not self.dag_acircle_check(start, p.io_owner.input_bindings.bindings):
+                            return False
+
+                return True
+
+            def connect(self, binding):
+                """
+                # check if the bindendency setting correct.
+                # correct connection are following
+                # 1. global input to module input
+                # 2. module output to global output
+                # 3. module output to moudle input
+                """
+                owner_indx = self.get_owner_indx()
+                bind_owner_indx = binding.get_owner_indx()
+                if owner_indx == bind_owner_indx:
+                    raise RuntimeError(f"can not set self as binding.")
+
+                if owner_indx != 0 and self.io_type == "input":
+                    raise RuntimeError(f"Module only can start binding from output!")
+
+                if owner_indx != 0 and bind_owner_indx != 0 and binding.io_type == "output":
+                    raise RuntimeError(f"Module output can not binding with module output!")
+
+                if owner_indx != 0 and bind_owner_indx == 0 and binding.io_type == "input":
+                    raise RuntimeError(f"Module output can not binding with global input!")
+
+                if owner_indx == 0 and self.io_type != "input":
+                    raise RuntimeError(f"Global only can start binding from input!")
+
+                if owner_indx == 0 and binding.io_type != "input":
+                    raise RuntimeError(f"Global input only can set binding with module input.")
+
+                self.bindings.append(binding)
+                if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
```

Comment:
This can also be `if not self.is_global_interface()`

Reply:
fixed.